docs/config_search.md

```diff
@@ -18,84 +18,132 @@ limitations under the License.
 
 Model Analyzer's `profile` subcommand supports multiple modes when searching to find the best model configuration.
 
-- [Brute](config_search.md#brute-search-mode) is the default when profiling non-ensemble models or when profling mulitple models sequentially, and will do a brute-force sweep of the cross product of all possible configurations
-- [Quick](config_search.md#quick-search-mode) is the default for ensemble models or when profiling multiple models concurrenty, and will use heuristics to try to find the optimal configuration much quicker than brute, and can be enabled via `--run-config-search-mode quick`
-- [Multi-model](config_search.md#multi-model-search-mode) will profile mutliple models to find the optimal configurations for all models while they are running concurrently. This feature is enabled via `--run-config-profile-models-concurrently-enable`
+- [Brute Force Search](config_search.md#brute-search-mode)
+  - `Search type:` Brute-force sweep of the cross product of all possible configurations
+  - `Default for:`
+    - Single non-ensemble models
+    - Multiple models being profiled sequentially
+  - `Command:` `--run-config-search-mode brute`
+- [Quick Search](config_search.md#quick-search-mode)
+  - `Search type:` Heuristic sweep using a hill-climbing algorithm to find an optimal configuration
+  - `Default for:`
+    - Single ensemble models
+    - Multiple models beging profiled concurrently
+  - `Command:` `--run-config-search-mode quick`
+- [Concurrent / Multi-model Search](config_search.md#multi-model-search-mode)
```

Comment:
I think having this section here is a bit confusing. It breaks the paradigm of brute force vs quick search, since concurrent search isn't an alternative to brute or quick (it's in a different category). While I like the layout of this, I think to minimize confusion, it might make sense to move this information, like to the sections you have below.

If you want to do a summary table like this at the top, I think a separate table with single-model non-ensemble, multi-model non-ensemble (sequential), multi-model non-ensemble (concurrent), and single-model ensemble could be useful and clear.

Reply:
Yeah, I'm not a fan of this either, but wanted to see what you two thought before I made major changes here.
My new plan is to have two sections: one for search types (quick vs. brute) and another for model types (single, sequential, concurrent, ensemble).